network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -252,4 +265,19 @@ public String toString() {
         .toString();
     }
   }
+
+  /**
+   * write out the set of registered executors to a file so we can reload them on restart.
+   * You must have a lock on executors when calling this
+   */
+  private void saveRegisteredExecutors() throws IOException {
+    if (registeredExecutorFile != null) {
+      File tmp = File.createTempFile("registeredExecutors",".bin");
```

Comment:
nit: space after `,`. Also, probably better if the temp file is created in the same directory as the destination (avoids non-atomic move if dirs are on different fses).